### ISSUE TITLE
Reading: A Visual Guide to Mixture Models

### ISSUE THREAD

**Issue by josephmaa** [ldoshi/rome-wasnt-built-in-a-day#250]
Link to article: https://newsletter.maartengrootendorst.com/p/a-visual-guide-to-mixture-of-experts

Main idea: 

- Multiple experts on different layers of a network can be used as a sub-component of a large language model.
- Experts are also generally feed forward neural networks
- The router network determines what expert to send a token to, per layer
- The mixture of expert models start with the feed-forward neural network, the tokens are densely connected in FFNNs.
- Softmax used to pick between different experts
- The routing model can be considered the most important part of any mixture of experts since it determines which expert will do training/inference
- Also an idea of adding some gaussian noise to the routing matrix so that the training won't select some experts over and over again.
- Load balancing for balancing expert importance
- Extra loss component involving importance

Mixture of Expert LLMs:

Link to article: https://cameronrwolfe.substack.com/p/moe-llms

- DeepSeek, Grok, Mixtral models use mixture of experts
- Example of a decoder only model
- Some idea of shared experts

For next time: Variational Neurons in Transformers for Language Modeling (2026): keeps the Transformer backbone but replaces feed-forward computation with local variational neurons.
